### DISCUSSION TITLE
Error during installation

### DISCUSSION THREAD

**Discussion by Joooonathan** [requarks/wiki#3103] [Help / Questions]
Hello all,

when installing wiki.js on my vServer I get the following error messages:

`wiki.service: Changing to the requested working directory failed: No such file or directory
wiki.service: Failed at step CHDIR spawning /usr/bin/node: No such file or directory
wiki.service: Main process exited, code=exited, status=200/CHDIR`

I don't know what the reason is.

The wiki.service file looks like this:
`[Unit]
Description=Wiki.js
After=network.target

[Service]
Type=simple
ExecStart=/usr/bin/node server
Restart=always
(#) Consider creating a dedicated user for Wiki.js here:
User=nobody
Environment=NODE_ENV=production
WorkingDirectory=/var/www/html/wiki

[Install]
WantedBy=multi-user.target`

During the installation I noticed that the command `node server` from the instructions does not work.
-bash: node: command not found

I ask for help as I cannot solve the problem myself.

Many thanks in advance.

Greetings

**Comment by mattpr** [comment]
> `wiki.service: Changing to the requested working directory failed: No such file or directory`

Just reading the error message...my guess is that the working directory you told systemd to use for this unit (`WorkingDirectory=/var/www/html/wiki`) does not exist.

> `-bash: node: command not found`

It also sounds like you have not installed nodejs or nodejs does not exist on the `PATH` being used by systemd.

> I ask for help as I cannot solve the problem myself.

I would respectfully point out that these are incredibly basic problems.  The fact that you are having trouble with them is a sign you will have a hard time setting up or maintaining the install longer term.  There are lots of more complex things that can easily go wrong.  Nevertheless, it is great you are trying.  The best way to learn is by "doing" (or trying and failing, learning and trying again).

Given these are "I'm new to linux/command-line" type problems and not wiki.js specific, you will probably get better/faster answers just googling the basic error messages and reading a bunch.  

**For instance searching:**

- `.service: Changing to the requested working directory failed: No such file or directory`
    - I removed the `wiki` from `wiki.service` since this is a general problem 
    - will likely lead you to lots of answers about systemd units created with working directories that don't exist.
- `node: command not found`
    - will likely lead you to lots of answers about installing nodejs and/or how to configure your `PATH` on linux.

I hope that helps!
